### PR TITLE
Timer Series Insights fix on colliding properties

### PR DESCRIPTION
**Pull request by mcardosos**
Fixes https://github.com/Azure/azure-rest-api-specs/issues/1683
Some definitions had colliding properties (definitions had property `properties`, which was also defined on another definition that was included with `allOf`). Now the swagger can be modeled.
PTAL @sandshadow

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by sandshadow** [comment]
@mcardosos I took a look at the C# code generated by autorest with these changes. There are two problems.

First, EventHubEventSourceCreateOrUpdateParameters now takes a "Kind" enum in the constructor. Previously Kind was set in a static constructor to "Microsoft.EventHub". It is now possible for someone to create a EventHubEventSourceCreateOrUpdateParameters with the wrong value for Kind. The same problem applies to IoTHubEventSourceCreateOrUpdateParameters. 

Second, EventHubEventSourceResource and IoTHubEventSourceResource have lost their EventSourceResourceId property. 

Of the two problems, the second is a bigger issue. EventSourceResourceId is a public property that all event sources that come from Azure will have. That's why both EventHubEventSourceResource and IoTHubEventSourceResource inherited AzureEventSourceProperties. The issue with Kind is pretty clunky (requiring a argument be passed that can only have one value), but doesn't break scenarios.

**Comment by mcardosos** [comment]
@sandshadow
Isn't `AzureEventSourceProperties` present in`EventHubEventSourceCommonProperties` and `IoTHubEventSourceCommonProperties` ?

**Comment by sandshadow** [comment]
@mcardosos 
Yes, Both EventHubEventSourceCommonProperties and IoTHubEventSourceCommonProperties have 
      "allOf": [
        {
          "$ref": "#/definitions/AzureEventSourceProperties"
        }

But it seems that autorest doesn't follow the inheritance chain when there are multiple "allOf" entries. I found the following comment in an old copy I have of the spec:
>      // Having one "allOf" indicates inheritance. Having multiple "allOf" only brings in the properties of > each derived definition,
>      // not its parent definitions. So using "allOf" to reference the > EventHubEventSourceCommonProperties is not enough to
>      // bring in the properties of AzureEventSourceProperties or EventSourceCommonProperties.

**Comment by mcardosos** [comment]
@sandshadow
Thanks for explaining :) Fixed.

File: specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/timeseriesinsights.json [hunk 1]
```diff
@@ -1250,10 +1250,6 @@
             "name": "Kind",
             "modelAsString": false
           }
-        },
-        "properties": {
-          "x-ms-client-flatten": true,
-          "$ref": "#/definitions/EventSourceCommonProperties"
```
Comment (dsgouda) [inline specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/timeseriesinsights.json]: You're removing the `properties` property which is a required property here. Semantic validation error.

File: specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/timeseriesinsights.json [hunk 2]
```diff
@@ -1366,10 +1362,6 @@
           "type": "string",
           "description": "The kind of the event source.",
           "enum": [ "Microsoft.EventHub", "Microsoft.IoTHub" ]
-        },
-        "properties": {
-          "x-ms-client-flatten": true,
-          "$ref": "#/definitions/EventSourceCommonProperties"
```
Comment (dsgouda) [inline specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/timeseriesinsights.json]: Same error as above

**Review by dsgouda — CHANGES_REQUESTED** [review]
Semantic validation errors, please confirm with the RPs if this is the change they want.

File: specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/timeseriesinsights.json (hunk 2 above)
Comment (mcardosos) [reply specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/timeseriesinsights.json]: Fixed

File: specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/timeseriesinsights.json (hunk 1 above)
Comment (mcardosos) [reply specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/2017-02-28-preview/timeseriesinsights.json]: Fixed

**Review by dsgouda — APPROVED** [review]
LGTM, I assume the RP has also signed off on this.

**Pull request merged**